cpp/tests/column/column_view_test.cpp

```diff
@@ -20,15 +20,70 @@
 #include <cudf/utilities/type_dispatcher.hpp>
 #include <tests/utilities/base_fixture.hpp>
 #include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
 #include <tests/utilities/cudf_gtest.hpp>
 #include <tests/utilities/type_list_utilities.hpp>
 #include <tests/utilities/type_lists.hpp>
 
 #include <thrust/sequence.h>
 #include <random>
 
+template <typename T, typename T2 = void>
+struct rep_type_t {
+  using type = void;
+};
+
+template <typename T>
+struct rep_type_t<T, typename std::enable_if<cudf::is_timestamp<T>()>::type> {
+  using type = typename T::duration::rep;
+};
+
 template <typename T>
-struct TypedColumnTest : public cudf::test::BaseFixture {
+struct ColumnViewAllTypesTests : public cudf::test::BaseFixture {
 };
 
-TYPED_TEST_CASE(TypedColumnTest, cudf::test::Types<int32_t>);
+TYPED_TEST_CASE(ColumnViewAllTypesTests, cudf::test::FixedWidthTypes);
+
+template <typename FromType, typename ToType>
+void do_logical_cast(cudf::column_view const& input)
+{
+  if (std::is_same<FromType, ToType>::value) {
+    // Cast to same type
+    auto output = input.logical_cast(input.type());
+    cudf::test::expect_columns_equal(output, input);
+  } else if (std::is_same<typename rep_type_t<FromType>::type, ToType>::value) {
+    // Cast to integer type and back to timestamp
+    cudf::data_type integer_type{cudf::type_to_id<ToType>()};
+    auto integer_view = input.logical_cast(integer_type);
+    auto output       = integer_view.logical_cast(input.type());
```

Comment:
I'm thinking of changing this to check equality with an integer column constructed with the same input values instead of casting back to the input type, and vice versa for the case below.